gatsby/src/specs/minfraud/2.0/utils.ts

```diff
@@ -0,0 +1,315 @@
+  string_email: (): string => 'user@example.com',
+  string_hostname: (): string => 'example.com',
+  string_ipv4: (): string => '198.51.100.42',
+  string_ipv6: (): string => '2001:0db8:5b96:0000:0000:426f:8e17:642a',
+  string_uuid: (): string => '3fa85f64-5717-4562-b3fc-2c963f66afa6',
+  /* eslint-enable @typescript-eslint/camelcase */
+};
+
+const primitive = (schema: SchemaObject): unknown => {
+  schema = objectify(schema);
+  const { type, format } = schema;
+
+  const fn = (primitives as any)[`${type}_${format}`]
+    || (primitives as any)[(type as string)];
+
+  if (typeof(fn) === 'function') {
+    return fn(schema);
+  }
+
+  return 'Unknown Type: ' + schema.type;
+};
+
+const deeplyStripKey = (
+  input: any,
+  keyToStrip: string,
+  predicate: (
+    a: any,
+    b: any
+  ) => boolean = (): boolean => true
+): Record<any, any> => {
+  if (
+    typeof input !== 'object'
+      || Array.isArray(input)
+      || input === null
+      || !keyToStrip
+  ) {
+    return input;
+  }
+
+  const obj = Object.assign({
+  }, input);
+
+  Object.keys(obj).forEach(k => {
+    if (k === keyToStrip && predicate(obj[k], k)) {
+      delete obj[k];
+      return;
+    }
+    obj[k] = deeplyStripKey(obj[k], keyToStrip, predicate);
+  });
+
+  return obj;
+};
+
+const sampleFromSchema = (schema: SchemaObject, config: any = {
+}): any => {
+  let { type } = schema;
+  const {
+    allOf,
+    additionalProperties,
+    example,
+    items,
+    properties,
+  } = schema;
+  const { includeReadOnly, includeWriteOnly } = config;
+
+  if (example !== undefined) {
+    return deeplyStripKey(example, '$$ref', (val: unknown) => {
+      // do a couple of quick sanity tests to ensure the value
+      // looks like a $$ref that swagger-client generates.
+      return typeof val === 'string' && val.indexOf('#') > -1;
+    });
+  }
+
+  if (allOf) {
+    const foo = {
+    };
+    allOf.map((item: any) => merge(foo, item));
+    return sampleFromSchema(foo, config);
+  }
+
+  if (!type) {
+    if (properties) {
+      type = 'object';
+    } else if (items) {
+      type = 'array';
+    } else {
+      return;
+    }
+  }
+
+  if (type === 'object') {
+    const props = objectify(properties);
+    const obj = {
+    };
+    for (const name in props) {
+      if ( props[name] && props[name].deprecated ) {
+        continue;
+      }
+      if ( props[name] && props[name].readOnly && !includeReadOnly ) {
+        continue;
+      }
+      if ( props[name] && props[name].writeOnly && !includeWriteOnly ) {
+        continue;
+      }
+      (obj as any)[name] = sampleFromSchema(props[name], config);
+    }
+
+    if ( additionalProperties === true ) {
+      (obj as any).additionalProp1 = {
+      };
+    } else if ( additionalProperties ) {
+      const additionalProps = objectify(additionalProperties);
+      const additionalPropVal = sampleFromSchema(additionalProps, config);
+
+      for (let i = 1; i < 4; i++) {
```

Comment:
Where does 4 come from?

Reply:
I took this `sampleFromSchema()` function from https://github.com/swagger-api/swagger-ui/blob/074eaa5d51b607bf5293d9519559b133d8295215/src/core/plugins/samples/fn.js#L34, and modified it for our needs.

I'll going to clean this up a bit. We don't have a testing suite in place now, but this is an area that we need to test. I'll add `TODO` comments to remind us.